tfjs-backend-webgpu/src/shader_util.ts

```diff
@@ -32,3 +32,20 @@ export function symbolicallyComputeStrides(
 
   return strides;
 }
+
+// atomicAdd only supports uint/int type. For float, we use
+// atomicCompareExchangeWeak to simulate.
+export const atomicAddSnippet = (dst: string, val: string) => {
```

Comment:
Can we also pass in type so that i32 and f32 can be both covered?
It's better to follow wgsl spec, and name the parameters as ptr and v, then we need to pass in address. 
With above, we can remove atomicRMW.